### ISSUE TITLE
ModItem.HoldItem + Main.mouseItem cloning behavior problem

### ISSUE THREAD

**Issue by direwolf420** [tModLoader/tModLoader#808]
### Description
Having a field on a ModItem, using CloneNewInstances => true, and doing something to the variables in the HoldItem hook, while having the item in the mouse cursor, causes the item to reset its values constantly.

### Steps to Reproduce
Tested on this item: https://paste.mod.gg/lavimigaxa.cs (Link expired, gist of the code below)

1. take item into the mouse
2. set a breakpoint in the `useCD--` line
3. wait until it hits, then set breakpoint on the `CloneNewInstances => true` line
4. Inspect useCD value of inventory[58].modItem vs Main.mouseItem.modItem when it hits the Clone() 
Cloning happens somewhere in dropItemCheck() here:
```cs
if (Main.myPlayer == this.whoAmI)
{
    this.inventory[58] = Main.mouseItem.Clone();
}
```

### Expected behavior 
HoldItem hook working properly for items held in the mouse

### Actual behavior
An "old" copy of Main.mouseItem being cloned into inventory[58]

### Reproduction frequency
100%

### Why this is important
Often times you use your weapons/items by taking them out of the inventory, clicking, then putting them back (For example if your hotbar is always filled with other things you need)

### Screenshots
https://cdn.discordapp.com/attachments/103115427491610624/685788000310919176/Screenshot_2.png
https://cdn.discordapp.com/attachments/103115427491610624/685788002307670198/Screenshot_4.png

### Gist of the code since the example expired
```cs
override CloneNewInstances => true;
public int useCD = 1000;
override HoldItem() => useCD--;

**Comment by Jofairden** [comment]
Is the part below step 4 what you suspect to be the culprit ?

**Comment by direwolf420** [comment]
No, this is just where cloning happens and where one of the screenshots is inspecting Main.mouseItem from.

**Comment by Chicken-Bones** [comment]
#1131 is removing `CloneNewInstances`

**Comment by Chicken-Bones** [comment]
As far as I can tell, this is fixed by #2398 

It's always 1000 for me.
![image](https://user-images.githubusercontent.com/388233/169899738-a50521f4-e674-4406-bfa4-8ca429393e9f.png)

**Comment by direwolf420** [comment]
Ok but this it NOT intended behavior. You having the item out in the cursor counts as "holding the item", and thus should decrement the variable each tick

**Comment by Chicken-Bones** [comment]
Probably something in vanilla then, cloning one way but not the other.

**Comment by Jofairden** [comment]
I thought we had clone specific hooks (or are they completely gone now?) and you havent copied the actual value of the cd in it? @direwolf420 @Chicken-Bones

**Comment by Chicken-Bones** [comment]
Sure but the default implementation is `MemberwiseClone` which should keep the counter value. My guess is the mouse item is a clone, and hooks run on that, but then it gets thrown away and the actual inventory item that you put back is unaffected.

**Comment by Chicken-Bones** [comment]
(mouse item is such a hack, why isn't it just an inv slot)

**Comment by Jofairden** [comment]
Actually, Im fairly certain at some point the mouse item is/was an inventory slot, the final slot that was hidden. I remember that the inventory array was 1 too big at first and I didn't know why. Did we remove that? @Chicken-Bones

**Comment by direwolf420** [comment]
both player.inventory[58] and Main.mouseItem exist and are treated as "item in players cursor", but in slightly different contexts, which is confusing. I haven't looked into this in a while though.

**Comment by Jofairden** [comment]
That does sound confusing, sound like it should get looked into.
